trunk/modules/python/dataset/reader.py

```diff
@@ -78,53 +83,109 @@ def openSICD(image, log=None) :
     wbData, cmplxData = pysix.six_sicd.read(image, schemaPaths)
     return (wbData, cmplxData)
 
-def readSICD(image, log=None) :
+def readSICD(image, log=None, **kwargs) :
```

Comment:
Add open* methods for each datatype, and keep the "raw" logic at a higher level. These methods should know nothing about the implementation of HDF5 or its inner-workings. Their only responsibility should be to returning the pixel data from an image on disk. 

I thought about this and it makes sense to have two methods. open* is for reading the raw data, and putting it in shape = (numChannel, numRow, numCol) format. For PIL this will also cast to floatX. However this set of open* will do no remapping of the data. The read format could then just use open* and then call the appropriate remap function on the np.ndarray that is returned. 

The readImage method could also just understand that np.complex64 should be remapped with convertToPhaseAmp() and all other data needs statisticalNorm().  This would allow the open* methods to get the read and sizing correct, and it would remove the need for all the per type read* methods.

Doing it in this manner would get you the datatype and formatting you need in the HDF5 Filesystem file, and judging by the dtype and shape, readImage could remap it appropriately. You won't have to store any additional data about the imagery, and it would remove all of the tuple logic you must carry around and account for in the current implementation.